charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml

```diff
@@ -65,6 +65,9 @@ spec:
             {{- end }}
             - --kubelet-endpoints={{ .Values.prometheusOperator.kubeletEndpointsEnabled }}
             - --kubelet-endpointslice={{ .Values.prometheusOperator.kubeletEndpointSliceEnabled }}
+            {{- if .Values.prometheusOperator.kubeletService.labels }}
```

Comment:
I'm confused, reading the code of the operator, this field is in no way limited to the kubelet service. It should affect all resources created by the operator right?

Reply:
I'm wondering if we could not just add an extraArgs for the operator because the value of `prometheusOperator.labels` is already taken 🤔 @jkroepke any thoughts?

Reply:
Now I am confused. According to the documentation you are absolutely correct. But I do not see such behavior. I see it only adds to kube-prometheus-kubelet service but no other and that is the only service seems to preserve even after uninstalling the helm chart. Double checked again now unless I am missing something.

![image](https://github.com/user-attachments/assets/8c2c9239-dc64-4571-beb6-d13ca0bf6b8d)

to kubelet

![image](https://github.com/user-attachments/assets/8e4b6007-3de5-489f-ba8f-f3060d192fdf)

Reply:
It should be added to Secrets as well, e.g. config secrets for AM and Prometheus. In context of services, the operations manages only the kubelet service.

Reply:
+1 for extraArgs option. 

Anyways, the use-case to put the helm release labels as options might be an interesting one in context of automatic cleanups. But I would expect that helm deletes the kubelet service + config secrets on each helm upgrade as well. Thats an dangerous approach.

Reply:
that deletion must be taken care by operator itself right. Helm has nothing to do with that.

Reply:
I see the same labels are being added to created secrets 

![image](https://github.com/user-attachments/assets/7f64bc08-7319-4fc4-b9cf-cec02e5b2f8d)

![image](https://github.com/user-attachments/assets/d05b10e0-170b-4fe1-94f4-004dbd017086)

![image](https://github.com/user-attachments/assets/e00c51a2-52ed-4117-b7c0-7758e463db5e)